llvm/lib/Target/AMDGPU/AMDGPUInsertDelayAlu.cpp

```diff
@@ -236,6 +251,17 @@ class AMDGPUInsertDelayAlu : public MachineFunctionPass {
       }
     }
 
+    void advanceByNum(DelayType Type, unsigned Cycles,
+                      unsigned SGPRWriteVALUNum) {
+      iterator Next;
+      for (auto I = begin(), E = end(); I != E; I = Next) {
+        Next = std::next(I);
+        if (I->second.VALUNum >= SGPRWriteVALUNum && I->second.VALUCycles > 0) {
+          erase(I);
+        }
+      }
+    }
```

Comment:
Type and Cycles aren't actually used, which makes this method misleading.

You could rename it to `advanceByVALUNum` and only pass the last argument.